scanpy/readwrite.py

```diff
@@ -410,7 +415,10 @@ def read_visium(
         }
 
         # read coordinates
-        positions = pd.read_csv(files['tissue_positions_file'], header=None)
+        positions = pd.read_csv(
+            files['tissue_positions_file'],
+            header=1 if tissue_positions_file.name == "tissue_positions.csv" else None,
+        )
         positions.columns = [
```

Comment:
There is one missing edit (which we fixed in squidpy https://github.com/scverse/squidpy/blob/fb069ded7515e8a1386224d32344a8657cbecd2e/squidpy/read/_read.py#L78-L89 ) that is, the order of the columns is different.

Reply:
@giovp this was actually a typo in the spaceranger documentation early on which has been fixed for some time. The order of the column names has always been `barcode,in_tissue,array_row,array_col,pxl_row_in_fullres,pxl_col_in_fullres`  no matter the version of spaceranger.

Reply:
Very sorry for the mixup! This should be updated in squidpy as well.